src/doom/hu_stuff.c

```diff
@@ -590,6 +590,8 @@ void HU_Start(void)
     const char *s;
     // [crispy] string buffers for map title and WAD file name
     char	buf[8], *ptr;
+    // [crispy] shift widgets one line down so chat typing line may appear
+    const int net_chatline = netgame ? 8 : 0;
```

Comment:
Could this be a bit more dynamic? For example, AV.wad plays a netgame demo even if you are going to play a singleplayer game. Could it get modified so that the widgets only slip down if there is an actual chat visible and shift back to their original positions else?

Reply:
Sure. I can abuse `HU_Ticker` with it's `netgame` condition, which will recalculate widgets position every calculated tic. To prevent duplication of `HUlib_initTextLine(&w_kills, ...` and others, I have to move them to static, separated function. Quick example how it will looks: https://youtu.be/3H2x0rUocXU

No commit for now, polishing and experementing is still needed.